### PR TITLE
Set up cspell (via Code Spell Checker extension)

### PR DESCRIPTION
**Pull request by andyw8**
### Motivation

Catch mistakes automatically. I have it installed locally, and I've been adding entries to my local dictionary, but I thought it would be beneficial to have as a shared dictionary.

(Partly addresses https://github.com/Shopify/ruby-dev-exp-issues/issues/590)

### Implementation

I used the command line `cspell` app to generate a list of the unknown words.

We can add this on other repos if folk like it.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,5 +3,11 @@
   "ruby lsp.trace.server": "messages",
   "[ruby]": {
     "editor.defaultFormatter": "Shopify.ruby-lsp",
-  }
+  },
+  "cSpell.languageSettings": [
+    { "languageId": "*",      "locale": "en",              "dictionaries": ["wordsEn"] },
+    { "languageId": "*",      "locale": "en-US",           "dictionaries": ["wordsEn"] },
+    { "languageId": "*",                                   "dictionaries": ["companies", "softwareTerms", "misc"] },
+    { "languageId": "ruby",                                "dictionaries": ["ruby"]},
```
Comment (andyw8) [inline .vscode/settings.json:11]: https://github.com/streetsidesoftware/cspell-dicts/blob/main/dictionaries/ruby/src/ruby.txt

File: .vscode/settings.json [hunk 2]
```diff
@@ -3,5 +3,11 @@
   "ruby lsp.trace.server": "messages",
   "[ruby]": {
     "editor.defaultFormatter": "Shopify.ruby-lsp",
-  }
+  },
+  "cSpell.languageSettings": [
```
Comment (andyw8) [inline .vscode/settings.json:7]: Based on the default from the README:
https://github.com/streetsidesoftware/vscode-spell-checker#miscellaneous-dictionaries

**Comment by andyw8** [comment]
(now approved in Kepler)

**Comment by adisonlampert** [comment]
Should we add something to the README about this so other devs know it exists if they want to set it up? (Is there anything devs need to do to set it up?)

**Review by adisonlampert — APPROVED** [review]

**Comment by andyw8** [comment]
@adisonlampert Thanks, I added some notes.

btw, there's a lot of [choices](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) for how to configure this extension, so we might alter it as we get to better understand it.

**Pull request merged**